### PR TITLE
eventLogger: redact repo, filenames and search queries from firstSourceURLs

### PR DESCRIPTION
**Pull request by attfarhan**
This updates our logic for tracking firstSourceURLs, which we use for marketing attribution. Currently, we will store the first sourcegraph URL a user has ever visited in a cookie, and associate that with their profile in HubSpot when they create one. However, with private code coming onto Cloud, we want to make sure we don't leak any private info like repo or file names, or the contents of search queries. 

This PR makes it so we replace these parts of the URL with `redacted`. For attribution, we only need the query parameters, which we maintain. So, with this change, we remove the risk of storing private information in a third-party tool, while ensuring our attribution remains effective.

### CODE REVIEW

File: client/web/src/tracking/util.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * @param url the original, full URL
+ */
+export function redactSensitiveInfoFromURL(url: string): string {
+    // Ensure we do not leak repo and file names in the URL
+    const sourceURL = new URL(url)
+    sourceURL.pathname = '/redacted'
+
+    // Ensure we do not leak search queries in the URL
+    const searchQuery = sourceURL.searchParams.get('q')
```
Comment (eseliger) [inline client/web/src/tracking/util.ts:27]: Can there be other sensitive things, too? This page for example https://k8s.sgdev.org/site-admin/repositories?filter=cloning&query=secret+repo

File: client/web/src/tracking/eventLogger.ts [hunk 2]
```diff
@@ -98,19 +99,23 @@ export class EventLogger implements TelemetryService {
...
-            expires: 365,
-            // Enforce HTTPS
-            secure: true,
-            // We only read the cookie with JS so we don't need to send it cross-site nor on initial page requests.
-            sameSite: 'Strict',
-            // Specify the Domain attribute to ensure subdomains (about.sourcegraph.com) can receive this cookie.
-            // https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#define_where_cookies_are_sent
-            domain: location.hostname,
-        })
+        {
```
Comment (eseliger) [inline client/web/src/tracking/eventLogger.ts:106]: Why this surrounding block?

**Review by eseliger — APPROVED** [review]

File: client/web/src/tracking/eventLogger.ts (hunk 2 above)
Comment (attfarhan) [reply client/web/src/tracking/eventLogger.ts:106]: Should be removed, thanks :)

File: client/web/src/tracking/util.ts (hunk 1 above)
Comment (attfarhan) [reply client/web/src/tracking/util.ts]: Yeah this is a good point. I guess it is possible for sensitive info to get leaked via search query parameters on arbitrary search bars/pages that we add. So, I think the approach should be "remove all query params except the marketing UTM params".

**Pull request merged**